### PR TITLE
Clean up Utf8CodeUnit and increase coverage

### PR DESCRIPTION
**Pull request by krwq**

### CODE REVIEW

**Comment by krwq** [comment]
@dotnet-bot test this please

File: src/System.Text.Utf8/src/System/Text/Utf8/Utf8CodeUnit.cs [hunk 1]
```diff
@@ -2,21 +2,26 @@
...
 namespace System.Text.Utf8
 {
-    // TODO: Should we directly use byte or leave it. If we leave it should we leave the name?
-    public struct Utf8CodeUnit : IEquatable<Utf8CodeUnit>
+    // TODO: Check if name is understandable for users
+    //       I personally think it is better to keep the spec name than calling it with some random new name or byte
+    //       In the end it is less confusing than trying to figure out what spec means and what .NET means
+    public partial struct Utf8CodeUnit : IEquatable<Utf8CodeUnit>
     {
+        private byte _value;
```
Comment (livarcocc) [inline src/System.Text.Utf8/src/System/Text/Utf8/Utf8CodeUnit.cs]: I couldn't see from the diff, why did you switch from the property to a private variable?

**Comment by livarcocc** [comment]
LGTM

File: src/System.Text.Utf8/src/System/Text/Utf8/Utf8CodeUnit.cs (hunk 1 above)
Comment (krwq) [inline src/System.Text.Utf8/src/System/Text/Utf8/Utf8CodeUnit.cs]: There was a perf issue about that which Ahson and Alex Perovich pointed out on some offline conversation. Compiler cannot treat a field as readonly or something like that (don't remember exact reason - I was told to change it)

**Pull request merged**